### PR TITLE
Enum plus struct

### PR DESCRIPTION
**Pull request by zancas**

### CODE REVIEW

**Review by AloeareV — APPROVED** [review]
Big fan of the new struct! I'm approving, but I'm not merging as both a) it hasn't passed CI yet, and b) I pushed a couple changes I'd like you to look at before it gets merged. If you approve of the changes, you're welcome to merge this!

File: zingolib/src/lightclient.rs [hunk 1]
```diff
@@ -1772,20 +1766,16 @@ impl LightClient {
...
                             kind: ValueTransferKind::Sent,
                             memos,
                             pool: None,
                             price,
                             to_address: Some(to_address),
                             txid,
                         });
-                        // We've accounted for the fee in the first send, but the fee
-                        // is for the full transaction
-                        fee = None;
```
[the remark this answers is not in the document]
Comment (zancas) [reply zingolib/src/lightclient.rs]: I think we need to separate balance calculation.

[the remark this answers is not in the document]
Comment (AloeareV) [reply zingolib/src/lightclient.rs]: Sure. I want `ValueTransferSummary` to be an accurate listing of all value transfers in the wallet, though... I haven't come up with anything better than having a `TransactionFeePayment` as a 4th variant yet, still thinking on it.

**Comment by zancas** [comment]
Superceded

**Pull request closed without merging**